development/webpack/utils/plugins/ManifestPlugin/helpers.ts

```diff
@@ -29,6 +35,48 @@ export function transformManifest(args: { lockdown: boolean; test: boolean }) {
     transforms.push(removeLockdown);
   }
 
+  /**
+   * This function sets predefined flags in the manifest's _flags property
+   * that are stored in the .manifest-flags.json file.
+   *
+   * @param browserManifest - The Chrome extension manifest object to modify
+   */
+  function addManifestFlags(browserManifest: chrome.runtime.Manifest) {
+    let manifestFlags;
+
+    if (manifestOverridesPath) {
+      try {
+        const fs = require('node:fs');
+        const path = require('node:path');
+        const manifestFlagsContent = fs.readFileSync(
+          path.resolve(process.cwd(), manifestOverridesPath),
+          'utf8',
+        );
+        manifestFlags = JSON.parse(manifestFlagsContent);
+      } catch (error: unknown) {
+        if (
+          error instanceof Error &&
+          'code' in error &&
+          error.code === 'ENOENT'
+        ) {
+          // Only throw if ENOENT and manifestOverridesPath was provided
+          throw new Error(
+            `Manifest override file not found: ${manifestOverridesPath}`,
+          );
+        }
+      }
+    }
+
+    if (manifestFlags) {
+      browserManifest._flags = manifestFlags;
```

Comment:
```suggestion
      browserManifest = merge(browserManifest, manifestFlags);
```

Also needs at the top of this file:
`import { merge } from 'lodash';`

Reply:
Addressed in ab7448dc61790c45092eb55b173ac86815618ec2. Also used `import merge from 'lodash/merge';` for optimal tree-shaking 🙏🏻

Reply:
@davidmurdoch something really confusing is happening here, it probably has to do with scope and transforms.

`return mergeWith(cloneDeep(browserManifest), manifestFlags);` has no effect

`return mergeWith(browserManifest, manifestFlags);` does the correct thing

If you console.log right there, either way seems to work, but only the bottom way adds stuff to the built manifest.

Reply:
No need to `cloneDeep`. `browserManifest` is _supposed_ to be mutated. So `addManifestFlags` shouldn't even have a return value.

Reply:
I tested, and this is working properly now.